src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -208,10 +272,12 @@ private ImmutableArray<Diagnostic> GetDiagnostics_NoLock(AnalysisScope analysisS
                 {
                     AddAllLocalDiagnostics_NoLock(_localSyntaxDiagnosticsOpt, analysisScope, builder);
                     AddAllLocalDiagnostics_NoLock(_localSemanticDiagnosticsOpt, analysisScope, builder);
+                    AddAllLocalDiagnostics_NoLock(_localAdditionalFileDiagnosticsOpt, analysisScope, builder);
                 }
                 else if (analysisScope.IsSyntaxOnlyTreeAnalysis)
                 {
                     AddLocalDiagnosticsForPartialAnalysis_NoLock(_localSyntaxDiagnosticsOpt, analysisScope, builder);
+                    AddLocalDiagnosticsForPartialAnalysis_NoLock(_localAdditionalFileDiagnosticsOpt, analysisScope, builder);
```

Comment:
The `if` above says `IsSyntaxOnlyTreeAnalysis`. Given that why would we be considering additional files here?

Reply:
I have renamed the flag to indicate this is for all single file syntactic analysis, source or additional file.